src/vs/workbench/contrib/chat/browser/chatSessions.contribution.ts

```diff
@@ -879,36 +880,40 @@ export class ChatSessionsService extends Disposable implements IChatSessionsServ
 				if (state.type !== IChatToolInvocation.StateKind.Completed) {
 					const pastTenseMessage = toolInvocation.pastTenseMessage;
 					const invocationMessage = toolInvocation.invocationMessage;
-					const message = pastTenseMessage || invocationMessage;
-					description = typeof message === 'string' ? message : message?.value ?? '';
+					description = pastTenseMessage || invocationMessage;
 
-					if (description) {
-						description = this.extractFileNameFromLink(description);
-					}
 					if (state.type === IChatToolInvocation.StateKind.WaitingForConfirmation) {
 						const message = toolInvocation.confirmationMessages?.title && (typeof toolInvocation.confirmationMessages.title === 'string'
 							? toolInvocation.confirmationMessages.title
 							: toolInvocation.confirmationMessages.title.value);
-						description = message ?? localize('chat.sessions.description.waitingForConfirmation', "Waiting for confirmation: {0}", description);
+						description = message ?? localize('chat.sessions.description.waitingForConfirmation', "Waiting for confirmation: {0}", typeof description === 'string' ? description : description.value);
 					}
 				}
 			}
 			if (!description && part.kind === 'toolInvocationSerialized') {
-				description = typeof part.invocationMessage === 'string' ? part.invocationMessage : part.invocationMessage?.value || '';
+				description = part.invocationMessage;
 			}
 			if (!description && part.kind === 'progressMessage') {
-				description = part.content.value || '';
+				description = part.content;
 			}
 		}
 
-		return description || localize('chat.sessions.description.working', "Working...");
+		return this.cleanUpProgress(description) || localize('chat.sessions.description.working', "Working...");
 	}
 
-	private extractFileNameFromLink(filePath: string): string {
-		return filePath.replace(/\[(?<linkText>[^\]]*)\]\(file:\/\/\/(?<path>[^)]+)\)/g, (match: string, _p1: string, _p2: string, _offset: number, _string: string, groups?: { linkText?: string; path?: string }) => {
-			const fileName = groups?.path?.split('/').pop() || groups?.path || '';
-			return (groups?.linkText?.trim() || fileName);
-		});
+	private cleanUpProgress(progress: IMarkdownString | string | undefined): string {
+		let cleaned = '';
+		if (typeof progress === 'string') {
+			cleaned = progress;
+		} else if (progress) {
+			// Remove file links but keep the link text or filename
+			cleaned = progress.value.replace(/\[(?<linkText>[^\]]*)\]\(file:\/\/\/(?<path>[^)]+)\)/g, (match: string, _p1: string, _p2: string, _offset: number, _string: string, groups?: { linkText?: string; path?: string }) => {
```

Comment:
We have a `renderAsPlaintext` helper that can strip out markdown syntax. Maybe what you are looking for here too?

Reply:
You can also use `marked.lexer` to parse the string and find links in a more reliable way

Reply:
<img width="331" height="188" alt="Screenshot 2025-11-26 at 6 51 46 PM" src="https://github.com/user-attachments/assets/913185ee-716b-4241-babb-7454487f6b0c" />
<img width="283" height="62" alt="Screenshot 2025-11-26 at 6 54 03 PM" src="https://github.com/user-attachments/assets/071e569e-1c2f-4180-9cf8-ba7b36176d49" />
@mjbvz using the string doesn't strip out the markdown link syntax and using the markdown just remove the link altogether :/

Reply:
I think `renderAsPlaintext` should work if you pass in a markdown string. You can create one if you need to:

```ts
console.log(renderAsPlaintext(new MarkdownString('[text](http://example.com)')));
// Logs 'text'
```

Reply:
However I'd look at why you have an empty link. That really would be invisible to the user if rendered directly as html